modules/infra/control/port.c

```diff
@@ -164,8 +164,18 @@ static int port_configure(struct iface_info_port *p) {
 	rxq_size = get_rxq_size(p, &info);
 	txq_size = get_txq_size(p, &info);
 
-	rte_mempool_free(p->pool);
-	p->pool = NULL;
+	mbuf_count = rxq_size * p->n_rxq;
+	mbuf_count += txq_size * p->n_txq;
+	mbuf_count += RTE_GRAPH_BURST_SIZE;
+	mbuf_count = rte_align32pow2(mbuf_count) - 1;
+	if (mbuf_count != p->mp_size) {
+		gr_pktmbuf_pool_release(p->pool, p->mp_size);
+		p->pool = gr_pktmbuf_pool_get(socket_id, mbuf_count);
+		p->mp_size = mbuf_count;
+	}
+
+	if (p->pool == NULL)
+		return errno_log(rte_errno, "gr_pktmbuf_pool_get");
```

Comment:
Maybe with the change in `gr_pktmbuf_pool_get`, use `errno` instead of `rte_errno`.